### PR TITLE
Adding blocking wait to shutdown API for timer queue in order to clea…

### PR DESCRIPTION
**Pull request by MushMal**
…r the executor thread invocation. Added unit tests.

*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by hassanctech — APPROVED** [review]
Looks good, let's wait for travis CI to succeed before merging though.

File: src/utils/tst/TimerQueue.cpp
Comment (lherman-cs) [inline src/utils/tst/TimerQueue.cpp:782]: This will always evaluate to true even in the bad path. I think we should add an acceptable to range to pass. Probably, [`curTime`, `curTime + 3 seconds + 300 ms (for the noise)`]?

**Review by lherman-cs — COMMENTED** [review]
Overall, it looks good to me. Only have 1 minor comment about the test.

Also, since we're changing the shutdown API to be blocking, would this break some of our higher level APIs? I'd imagine it should be working fine if we always call freeTimerQueue right after shutdown. But, I'm more concerned with the path where we call shutdown, and has another shutdown sequence from other parts before we call freeTimerQueue. The other shutdown sequences might wait the timerQueue to finish first.

File: src/utils/tst/TimerQueue.cpp
Comment (MushMal) [reply src/utils/tst/TimerQueue.cpp:782]: Oh, of course.

Comment (MushMal) [reply src/utils/tst/TimerQueue.cpp:782]: I am also thinking about the case where an API can be called after the shutdown. Need to add a check and a few tests.

Will be back with a quick change

**Review by lherman-cs — APPROVED** [review]
LGTM!

**Pull request merged**
